### PR TITLE
frontend: changed to eslint, removed deprecated tslint

### PR DESCRIPTION
**Pull request by thisconnect**
Removed tslint and configure eslint to lint js and ts files.
Added typescript-eslint and eslint react  modules and updated
eslint to latest, and a newer preact patch version.

Copied all rules from eslint-config-synacor and removed the
dependency, (wasnt updated in a year). The eslint rules now
contain the synacor merged with our own rules, marked with Shift.

Disabled all rules that warned or errored, so we can enable those
one by one in separate pull requests.

### CODE REVIEW

File: frontends/web/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    "no-duplicate-imports": "off",
+    // "no-else-return": 2,
+    "no-else-return": "off",
+    "no-empty-pattern": 0,
+    "no-empty": 0,
+    "no-extra-parens": 0, // we use extra parens in JSX ternaries, no way to disable that here.
+    "no-iterator": 2,
+    // "no-lonely-if": 2,
+    "no-lonely-if": "off",
+    "no-mixed-spaces-and-tabs": [2, "smart-tabs"],
```
Comment (benma) [inline frontends/web/.eslintrc.js:100]: i'd comment this for now, but do without 'smart-tabs' later. seems like smart-tabs allows pretty much everything.

File: frontends/web/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    "@typescript-eslint/no-unsafe-member-access": "off",
+    "@typescript-eslint/no-floating-promises": "off",
+    "@typescript-eslint/ban-types": "off",
+    "@typescript-eslint/ban-ts-comment": "off",
+    "@typescript-eslint/no-explicit-any": "off",
+    "@typescript-eslint/unbound-method": "off",
+    "@typescript-eslint/no-non-null-assertion": "off",
+    "@typescript-eslint/no-unnecessary-type-assertion": "off",
+    "@typescript-eslint/restrict-template-expressions": "off",
+    "@typescript-eslint/no-empty-interface": "off",
```
Comment (benma) [inline frontends/web/.eslintrc.js:192]: i hope we can keep this one disabled :) i like `interface Props {}` better, so it is named, than inlining `()` everywhere.

File: frontends/web/package.json [hunk 3]
```diff
@@ -6,43 +6,39 @@
     "gen-ts-css": "tcm src --camelCase dashes",
     "build": "yarn run gen-ts-css && preact build --no-sw --no-esm --no-inline-css --no-prerender --verbose",
     "dev": "yarn run gen-ts-css && preact watch",
-    "lint": "yarn run gen-ts-css && eslint --max-warnings 0 --cache --ignore-path ../../.gitignore '**/*.{js,jsx}'",
-    "postlint": "yarn run gen-ts-css && tslint -e 'src/**/*.css.d.ts' 'src/**/*.ts?(x)'",
+    "prelint": "yarn run gen-ts-css",
+    "lint": "eslint --max-warnings 0 --cache --ignore-path ../../.gitignore 'src/**/*.{js,jsx,ts,tsx}'",
```
Comment (benma) [inline frontends/web/package.json:10]: Would you mind adding `--format unix`? This is the format in which every editor can automatically jump to the source code at the error, as it is always `filename:linenumber <error message>`.

**Review by benma — APPROVED** [review]
nice!

File: frontends/web/.eslintrc.js (hunk 2 above)
Comment (thisconnect) [reply frontends/web/.eslintrc.js]: 👍  moved it from TODO to the Shift section

File: frontends/web/.eslintrc.js (hunk 1 above)
Comment (thisconnect) [reply frontends/web/.eslintrc.js:100]: https://github.com/digitalbitbox/bitbox-wallet-app/pull/1013

**Pull request merged**